### PR TITLE
Add Management Groups 2018-03-01-preview version

### PR DESCRIPTION
**Pull request by rajshah11**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @rajshah11 - could you take a look at the model validation errors:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/378549766

**Comment by annatisch** [comment]
@ravbhatnagar - looks like we have a new preview API for you to review :)
FYI @rajshah11

**Comment by ravbhatnagar** [comment]
Looks good.

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 1]
```diff
@@ -0,0 +1,1500 @@
...
+      },
+      "x-ms-parameter-location": "client"
+    },
+    "SkipParameter": {
+      "name": "$skip",
+      "in": "query",
+      "required": false,
+      "type": "integer",
+      "description": "Number of entities to skip over when retrieving results. Passing this in will override $skipToken.",
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 2]
```diff
@@ -0,0 +1,1500 @@
...
+    },
+    "CheckNameAvailabilityParameter": {
+      "name": "checkNameAvailabilityRequest",
+      "in": "body",
+      "required": true,
+      "description": "Management group name availability check parameters.",
+      "schema": {
+        "$ref": "#/definitions/CheckNameAvailabilityRequest"
+      },
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 3]
```diff
@@ -0,0 +1,1500 @@
...
+      "name": "operationResultId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "create",
+        "delete"
+      ],
+      "description": "The id of the operation result.",
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 4]
```diff
@@ -0,0 +1,1500 @@
...
+    },
+    "PatchGroupRequestParameter": {
+      "name": "patchGroupRequest",
+      "in": "body",
+      "required": true,
+      "description": "Management group patch parameters.",
+      "schema": {
+        "$ref": "#/definitions/PatchManagementGroupRequest"
+      },
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 5]
```diff
@@ -0,0 +1,1500 @@
...
+    },
+    "CreateManagementGroupRequestParameter": {
+      "name": "createManagementGroupRequest",
+      "in": "body",
+      "required": true,
+      "description": "Management group creation parameters.",
+      "schema": {
+        "$ref": "#/definitions/CreateManagementGroupRequest"
+      },
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 6]
```diff
@@ -0,0 +1,1500 @@
...
+      "description": "Number of entities to skip over when retrieving results. Passing this in will override $skipToken.",
+      "x-ms-parameter-location": "client"
+    },
+    "TopParameter": {
+      "name": "$top",
+      "in": "query",
+      "required": false,
+      "type": "integer",
+      "description": "Number of elements to return when retrieving results. Passing this in will override $skipToken.",
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 7]
```diff
@@ -0,0 +1,1500 @@
...
+    },
+    "SkipTokenParameter": {
+      "name": "$skiptoken",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "description": "Page continuation token is only used if a previous operation returned a partial result. \nIf a previous response contains a nextLink element, the value of the nextLink element will include a token parameter that specifies a starting point to use for subsequent calls.\n",
+      "x-ms-parameter-location": "client"
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: The parameter location needs to be "method"

**Review by annatisch — CHANGES_REQUESTED** [review]
Some small fixes are needed:
- please remove all the newline characters from the end of descriptions.
- please make query parameters located on the method rather than client.
- please fix the last model validation error in the CI

File: specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json [hunk 8]
```diff
@@ -0,0 +1,1494 @@
...
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "GetEntities": {
+            "$ref": "./examples/GetEntities.json"
+          }
+        },
+        "x-ms-long-running-operation": true
```
Comment (annatisch) [inline specification/managementgroups/resource-manager/Microsoft.Management/preview/2018-03-01-preview/management.json]: @rajshah11 - can you confirm that this list operation is indeed also a long running operation?

**Review by annatisch — APPROVED** [review]

**Pull request merged**